typingserver/input/input.go

```diff
@@ -57,15 +57,21 @@ func (t *TypingServerInputAPI) InputTypingEvent(
 }
 
 func (t *TypingServerInputAPI) sendEvent(ite *api.InputTypingEvent) error {
-	userIDs := t.Cache.GetTypingUsers(ite.RoomID)
 	ev := &api.TypingEvent{
 		Type:   gomatrixserverlib.MTyping,
 		RoomID: ite.RoomID,
 		UserID: ite.UserID,
+		Typing: ite.Typing,
 	}
 	ote := &api.OutputTypingEvent{
-		Event:       *ev,
-		TypingUsers: userIDs,
+		Event: *ev,
+	}
+
+	if ev.Typing {
+		expireTime := ite.OriginServerTS.Time().Add(
+			time.Duration(ite.Timeout) * time.Millisecond,
```

Comment:
I wonder: should the timeout be something that is attached to every typing event, or should it be a global constant?

Reply:
Certainly looks quite good. I just have a few questions and suggestions.

Reply:
The client can either include or omit the `timeout` field when sending a typing event. The server only applies a default value when the field is omitted.

Reply:
Ah, yes I see. Missed that bit of the spec.